src/community/backup-restore/core/src/main/resources/applicationContext.xml

```diff
@@ -0,0 +1,798 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- <!DOCTYPE beans PUBLIC "-//SPRING//DTD BEAN//EN" "http://www.springframework.org/dtd/spring-beans.dtd"> -->
+<!-- Copyright (C) 2015 - 2016 - Open Source Geospatial Foundation. All rights 
+	reserved. This code is licensed under the GPL 2.0 license, available at the 
+	root application directory. -->
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+	xmlns:batch="http://www.springframework.org/schema/batch"
+	xmlns:util="http://www.springframework.org/schema/util"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="
+		http://www.springframework.org/schema/beans
+        http://www.springframework.org/schema/beans/spring-beans-4.0.xsd
+        http://www.springframework.org/schema/batch
+        http://www.springframework.org/schema/batch/spring-batch-3.0.xsd
+        http://www.springframework.org/schema/util
+        http://www.springframework.org/schema/util/spring-util-4.0.xsd">
+
+	<!-- core backup/restore setup -->
+
+	<import resource="backup-batch-context.xml" />
+
+	<bean id="backupFacade" class="org.geoserver.backuprestore.Backup">
+		<constructor-arg ref="catalog" />
+		<constructor-arg ref="resourceLoader" />
+		
+		<property name="totalNumberOfBackupSteps" value="9" />
```

Comment:
This looks like a derived property that could be counted from the backup and restore job definitions... if I understand it correctly, it seems that if a step needs adding, then this also has to be maintained.